engine/framework/internal/master/worker_manager.go

```diff
@@ -569,6 +578,25 @@ func (m *WorkerManager) checkMasterEpochMatch(msgEpoch frameModel.Epoch) (ok boo
 	return true
 }
 
+func (m *WorkerManager) checkWorkerEpochMatch(curEpoch, msgEpoch frameModel.Epoch) bool {
+	if msgEpoch > curEpoch {
+		log.Panic("We are a stale master still running",
+			zap.String("master-id", m.masterID), zap.Int64("own-epoch", m.epoch),
+			zap.Int64("own-worker-epoch", curEpoch),
+			zap.Int64("msg-worker-epoch", msgEpoch),
```

Comment:
- Is there a unit test case for this branch?
- Could panicking here make the system more unstable in production?

Reply:
This case may happen because of
- worker manager maintains a smaller worker epoch than that it uses to create worker
- a stale master, another job master creates a job and worker sends heartbeat to stale master

Both of them are bugs and should not happen.

Besides when job master panics here, new job master will load worker from metastore and uses correct worker epoch, it won't continue panic.